### PR TITLE
Encrypt cookies, pass UA to emulate webview

### PR DESCRIPTION
**Pull request by georgecrawford**
Not yet ready for merge - requires https://github.com/Financial-Times/newsletter-signup/pull/18 to be released and imported.

### CODE REVIEW

File: src/encryption.js [hunk 1]
```diff
@@ -0,0 +1,18 @@
+// See https://github.com/chris-rock/node-crypto-examples
+const crypto = require('crypto');
```
Comment (quarterto) [inline src/encryption.js]: `import`/`export` please

File: src/encryption.js [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+const algorithm = 'aes-256-ctr';
+const password = process.env.ENCRYPTION_PASSWORD;
+
+module.exports.encrypt = text => {
+	const cipher = crypto.createCipher(algorithm, password);
+	let crypted = cipher.update(text, 'utf8', 'hex');
```
Comment (quarterto) [inline src/encryption.js]: I could see hex strings getting quite long, maybe base64 instead?

File: src/encryption.js [hunk 3]
```diff
@@ -0,0 +1,19 @@
+import crypto from 'crypto';
+
+const algorithm = 'aes-256-ctr';
+const password = process.env.ENCRYPTION_PASSWORD;
+
+// See https://github.com/chris-rock/node-crypto-examples
+module.exports.encrypt = text => {
```
Comment (quarterto) [inline src/encryption.js]: hate to nitpick but `export` as well please (can't find an emoji that expresses "entreaty"). you might need to change the import in index.js. Option 1 is `export default { encrypt(...) {...}, decrypt(...) {...} }` and keep the import as-is, option 2 is `export const encrypt ... export const decrypt` and `import {encrypt, decrypt} from './encryption'` or `import * as encryption from './encryption'`. I'd prefer option 2.

**Comment by quarterto** [comment]
lgtm

**Comment by georgecrawford** [comment]
Thanks Matt

File: src/index.js [hunk 4]
```diff
@@ -94,38 +95,56 @@ app.get('/', (req, res) => {
...
+			query: {
+				...currentUrl.query,
+				external: true,
+				encryptedCookies: encrypt(req.get('cookie')),
+				ua: req.get('user-agent'),
+			},
+		});
+
+		if(app.locals.metadata) {
+			res.locals.metadata = JSON.stringify(JSON.parse(app.locals.metadata), null, 2);
```
Comment (quarterto) [inline src/index.js]: whoops, this is redundant now @georgecrawford

Comment (georgecrawford) [inline src/index.js]: https://github.com/Financial-Times/distro-light-signup/pull/38

**Pull request merged**